tests/controls/itemcontainertest.cpp

```diff
@@ -280,13 +280,27 @@ void ItemContainerTestCase::SelectionAfterDelete()
     container->Append("item 1");
     container->Append("item 2");
     container->Append("item 3");
+    container->Append("item 4");
+    container->Append("item 5");
+
+    // Selected item (not the last one) is after deleted item
+    container->SetSelection(4);
+    container->Delete(3);
+    CHECK( container->GetSelection() == wxNOT_FOUND );
```

Comment:
@vadz
What should be returned by wxItemContainer::GetSelection() is this case?
Documentation says that in this case current selection is invalidated so `wxNOT_FOUND` would be expected but tons of tests fail with this assertion.
For example for wxListBox (ListBoxTestCase) where wxItemConatiner is implemented as a native control in all major ports, returned selection index is 3 (initial decreased by one) for wxMSW, wxGTK2, wxOSX. For wxGTK3 it would probably return 3 too but this selection is invalidated explicitly in the code. So it looks selection is not invalidated in native controls.
When in this assertion expected selection index is set to 3 many other tests fail (expecting wxNOT_FOUND here) - probably those with non-native implementation of wxItemContainer, like e.g. HtmlListBoxTestCase
So should wxItemContainer::Delete() invalidate current selection in this case or not?

Reply:
I think we have a number of tickets about inconsistent behaviour of various controls but unfortunately I can't find them right now (I found https://trac.wxwidgets.org/ticket/19180 and https://trac.wxwidgets.org/ticket/18871 but I'm pretty sure there were more).

The problem is that, AFAIR, different controls currently behave in different ways and at least some of them behave in the same way under all platforms (or at least under MSW and GTK), so we can't even justify breaking compatibility for them by saying that we they work the same under all platforms in the new version even if they work differently than before. Instead each case has to be examined individually and I never had the time to do it...

Personally adjusting the selection makes much more sense to me than deleting it in this case (although deleting it would be logical if `Delete(4)` were called, of course), and it looks like we really need to do this for `wxListBox` in wxGTK3. But what about the other controls, including those not inheriting from `wxItemContainer`, such as `wx{List,Tree,DataView}Ctrl` that are rather similar but often end up behaving inconsistently?

It would be great to do something about this before 3.2.0, but unfortunately I still have too many other critical things to do I want to do until then myself, so I'm afraid I can't be of much help here. If you can work on this, I'd recommend starting by making a table comparing the behaviour of various controls under the 3 platforms in this case (and maybe also in case of `Insert()`-ing an item before/at the selected position) to see what would be the least amount of incompatible changes we could make to make things behaves consistently under all platforms. If you can do this, please post to wx-dev so that we could discuss it there.

TIA!